clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/MigrateLoadBalancerStrategy.java

```diff
@@ -0,0 +1,452 @@
+          .withVpcId(target.getVpcId())
+      ).getGroupId();
+      if (source.getVpcId() == null) {
+        addClassicLinkIngress(target, elbGroupId);
+        addPublicIngress(targetAmazonEC2, elbGroupId, sourceDescription);
+      }
+    }
+    return elbGroupId;
+  }
+
+  /**
+   * Creates the app specific security group, or returns the ID of one if it already exists
+   *
+   * @param appGroups  list of existing security groups in which to look for existing app security group
+   * @param target     the target location of the load balancer
+   * @param appName    the name of the application being migrated
+   * @param result     the result set for the load balancer migration - this will potentially be mutated as a side effect
+   * @param dryRun     whether the migration should actually occur or just be calculated
+   * @param elbGroupId the groupId of the elb specific security group, which will allow ingress permission from the
+   *                   app specific security group
+   */
+  protected void buildApplicationSecurityGroup(List<SecurityGroup> appGroups, LoadBalancerLocation target, String appName, MigrateLoadBalancerResult result, boolean dryRun, String elbGroupId) {
+    if (getDeployDefaults().getAddAppGroupToServerGroup()) {
+      AmazonEC2 targetAmazonEC2 = getAmazonClientProvider().getAmazonEC2(target.getCredentials(), target.getRegion(), true);
+      boolean exists = appGroups.stream().anyMatch(g -> g.getVpcId().equals(target.getVpcId()) && g.getGroupName().equals(appName));
+      if (!exists) {
+        MigrateSecurityGroupReference appGroupReference = new MigrateSecurityGroupReference();
+        appGroupReference.setAccountId(target.getCredentials().getAccountId());
+        appGroupReference.setVpcId(target.getVpcId());
+        appGroupReference.setTargetName(appName + "-elb");
+        MigrateSecurityGroupResult addedGroup = new MigrateSecurityGroupResult();
+        addedGroup.setTarget(appGroupReference);
+        result.getSecurityGroups().add(addedGroup);
+        if (!dryRun) {
+          String newGroupId = targetAmazonEC2.createSecurityGroup(
+            new CreateSecurityGroupRequest(appName, "Application security group for " + appName)
+              .withVpcId(target.getVpcId())
+          ).getGroupId();
+          OperationPoller.retryWithBackoff(
+            o -> appGroups.addAll(targetAmazonEC2.describeSecurityGroups(
+              new DescribeSecurityGroupsRequest().withGroupIds(newGroupId)).getSecurityGroups()),
+            200, 5);
+        }
+      }
+      SecurityGroup appGroup = appGroups.stream().filter(g -> g.getVpcId().equals(target.getVpcId()) && g.getGroupName().equals(appName)).findFirst().get();
+      if (!dryRun && appGroup.getIpPermissions().stream()
+        .noneMatch(p -> p.getUserIdGroupPairs().stream().anyMatch(u -> u.getGroupId().equals(elbGroupId)))) {
+
+        IpPermission newPermission = new IpPermission().withIpProtocol("tcp").withFromPort(7001).withToPort(7002)
```

Comment:
it would be better to use the ports from the ELB listener rather than 7001,7002 here (the `listener.instancePort`)